testing-modules/testing-libraries-2/src/test/java/com/baeldung/environmentvariablesfortest/EnvironmentVariablesSetDirectlyUnitTest.java

```diff
@@ -0,0 +1,31 @@
+package com.baeldung.environmentvariablesfortest;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.lang.reflect.Field;
+import java.util.Map;
+import static org.assertj.core.api.Assertions.assertThat;
+
+// This test can only work well in Java 15 and below
```

Comment:
Nice. I like it.

Reply:
Hmm, I am confused now, since CI runs this module on Java 17, the build should have failed previously for one of these tests but it did not. Can you please investigate running this test on Java 17 and understand how it is passing when you expected it to fail please?

Reply:
And also can you please clarify which of the two tests you expected to fail on Java 16+?